fairseq/tasks/speech_to_speech.py

```diff
@@ -343,14 +395,26 @@ def build_generator(
                 else self.vocoder.cpu()
             )
 
+        from fairseq.models.speech_to_speech import (
+            Translatotron2ConformerModel,
+            UnitYConformerModel,
+        )
+
         if self.args.target_is_code:
             if self.args.n_frames_per_step == 1:
-                seq_generator = super().build_generator(
-                    models,
-                    args,
-                    seq_gen_cls=None,
-                    extra_gen_cls_kwargs=extra_gen_cls_kwargs,
-                )
+                if isinstance(models[0], UnitYConformerModel):
```

Comment:
Does only `UnitYConformerModel` need `MultiDecoderSequenceGenerator`? How about other models with dual-decoder style architectures? Maybe it's easier to have something like `self.has_dual_decoder` in this task, which we can set the value when we read and parse `args.multitask_config_yaml`.